### PR TITLE
Move data anonymization to leaf subqueries.

### LINKED ISSUE #320: Move anonymization to leaf subqueries.
There are too many features we don't support yet, which makes anonymizing in the top query unfeasible.
Leaf subqueries referencing sensitive table should be the anonymizing ones.
That means no joins, no filters, no unsupported bucketization. Everything post-anonymization should be allowed (including ORDER BY, LIMIT, OFFSET, SELECT with grouping, and HAVING in the anonymizing subquery).

### PR DESCRIPTION
**Pull request by cristianberneanu**
First, the global execution context is dropped and replaced with a global query context plus a per-subquery node anonymization context. Afterwards, the limited support for AID values floating is also dropped and data anonymization is moved to the leaf subquery nodes.

Closes #320.

### CODE REVIEW

File: src/OpenDiffix.CLI/JsonEncodersDecoders.fs [hunk 1]
```diff
@@ -27,7 +27,6 @@ module private rec Encoders =
     | IntegerType -> "integer"
     | RealType -> "real"
     | StringType -> "text"
-    | ListType itemType -> typeName itemType + "[]"
```
Comment (edongashi) [inline src/OpenDiffix.CLI/JsonEncodersDecoders.fs:30]: Function `rec` decorator can be removed.

File: src/OpenDiffix.Core.Tests/Expression.Tests.fs [hunk 2]
```diff
@@ -535,8 +535,15 @@ let colRef2 = ColumnReference(2, RealType)
...
+let aggContext =
+  {
+    AnonymizationParams = AnonymizationParams.Default
+    GroupingLabels = [||]
+    Aggregators = [||]
+  }
+
 let evaluateAggregator aggSpec args =
-  evaluateAggregator (ExecutionContext.makeDefault ()) aggSpec args testRows
+  evaluateAggregator (aggContext, Some { BucketSeed = 0UL }) aggSpec args testRows
```
Comment (edongashi) [inline src/OpenDiffix.Core.Tests/Expression.Tests.fs:546]: Potentially dumb question, but can we make the `Some { BucketSeed = 0UL }` part of `aggContext`? Any specific reason to pass them around separately?

Comment (cristianberneanu) [reply src/OpenDiffix.Core.Tests/Expression.Tests.fs:546]: The two components of this context have different scopes: the aggregation context applies to the current aggregation operation and the anonymization context applies to the current bucket. It seems natural to keep them separate. In the case of evaluating aggregators, we could create another context structure to contain both components, but that adds another level of indirection and pollutes the type namespace. I wasn't convinced it is worth doing until now.

Comment (cristianberneanu) [reply src/OpenDiffix.Core.Tests/Expression.Tests.fs:546]: The "AnonymizationContext" type used here should be replaced with a "BucketContext" type, to make it more clear it applies to each individual bucket. The two are identical for now (they might not be in the future), so I postponed adding another type to reduce type pollution. But maybe that wasn't the best decision...

Comment (edongashi) [reply src/OpenDiffix.Core.Tests/Expression.Tests.fs:546]: We already have a ton of "context"s around so the simpler the better!

**Review by edongashi — APPROVED** [review]

**Pull request merged**